tests/test_git.py

```diff
@@ -1,12 +1,123 @@
 """This module contains the tests for the git container, the image with git pre-installed.
 """
-from bci_tester.data import GIT_CONTAINER
+import pytest
+from pytest_container.container import container_from_pytest_param
+from pytest_container.container import DerivedContainer
+from pytest_container.pod import Pod
+from pytest_container.pod import PodData
 
+from bci_tester.data import GIT_CONTAINER
 
 CONTAINER_IMAGES = (GIT_CONTAINER,)
 
+PRIV_KEY = """-----BEGIN OPENSSH PRIVATE KEY-----
+b3BlbnNzaC1rZXktdjEAAAAABG5vbmUAAAAEbm9uZQAAAAAAAAABAAAAMwAAAAtzc2gtZW
+QyNTUxOQAAACDJMR7NW+gNZGborupz8XoZEjuKRuKjLzVPAwfPcjrTzQAAAIi/QgI2v0IC
+NgAAAAtzc2gtZWQyNTUxOQAAACDJMR7NW+gNZGborupz8XoZEjuKRuKjLzVPAwfPcjrTzQ
+AAAEBACrN2+98i3BPX40CQxih8gRePIokGrmrobXVnNja+XckxHs1b6A1kZuiu6nPxehkS
+O4pG4qMvNU8DB89yOtPNAAAAA0JDSQEC
+-----END OPENSSH PRIVATE KEY-----"""
+
+PUB_KEY = "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIMkxHs1b6A1kZuiu6nPxehkSO4pG4qMvNU8DB89yOtPN BCI"
+
+GIT_SERVER_CONTAINERFILE = rf"""
+RUN zypper -n in git openssh-server
+RUN ssh-keygen -A
+RUN useradd -U -m -p "*" git
+
+RUN mkdir /home/git/.ssh && touch /home/git/.ssh/authorized_keys
+RUN chmod 700 /home/git/.ssh && chmod 600 /home/git/.ssh/authorized_keys
+RUN chown -R git:git /home/git/.ssh
+RUN echo "{PUB_KEY}" >> /home/git/.ssh/authorized_keys
+
+RUN mkdir -p /srv/git/project.git
+RUN git -C /srv/git/project.git init --bare
+
+RUN echo -e '\
+AuthorizedKeysFile      .ssh/authorized_keys\n\
+ChallengeResponseAuthentication no\n\
+PasswordAuthentication no\n\
+PermitRootLogin no\n\
+PubkeyAuthentication yes\n\
+AllowUsers git\n\
+UsePAM no\n\
+AllowTcpForwarding no\n\
+X11Forwarding no\n\
+Subsystem       sftp    /usr/lib/ssh/sftp-server\n\
+' > /etc/ssh/sshd_config
+
+EXPOSE 22
+
+CMD ["/usr/sbin/sshd", "-De"]
+"""
+
 
 def test_git_version(auto_container):
     assert "git version 2." in auto_container.connection.check_output(
         "git --version"
     )
+
+
+def test_git_clone_https(auto_container_per_test):
+    """Test that we can clone a repository over HTTPS successfully."""
+    auto_container_per_test.connection.run_expect(
+        [0], "git clone https://github.com/github/gemoji.git /gemoji"
+    )
+    auto_container_per_test.connection.file("/gemoji/.git/HEAD").exists
+    auto_container_per_test.connection.file("/gemoji/Gemfile").exists
+    auto_container_per_test.connection.file("/gemoji/db/emoji.json").exists
+    out = auto_container_per_test.connection.check_output(
+        "git -C /gemoji status"
+    )
+    assert "Your branch is up to date" in out
+    assert "nothing to commit, working tree clean" in out
+
+
+GIT_SERVER_CONTAINER = DerivedContainer(
+    base="registry.suse.com/bci/bci-base:latest",
+    containerfile=GIT_SERVER_CONTAINERFILE,
+)
+
+GIT_POD = Pod(
+    containers=[
+        GIT_SERVER_CONTAINER,
+        container_from_pytest_param(GIT_CONTAINER),
+    ],
+)
```

Comment:
There is a subtle issue here: the `GIT_POD` is now no longer using the marks from `GIT_CONTAINER`. QA is relying on these to filter test cases. So it will have to be something like this:
```suggestion
_ctr, _marks = container_and_marks_from_pytest_param(GIT_CONTAINER)
GIT_POD = pytest.param(Pod(
    containers=[GIT_SERVER_CONTAINER, _ctr],
), marks=_marks)
```